core/src/com/unciv/logic/city/CityInfo.kt

```diff
@@ -33,6 +36,7 @@ class CityInfo {
     @Transient var hasJustBeenConquered = false  // this is so that military units can enter the city, even before we decide what to do with it
 
     var location: Vector2 = Vector2.Zero
+    var id: String = ""
```

Comment:
if you put ` UUID.randomUUID().toString()` instead of empty string here you can remove setter from constructor and remove setter from GameInfo line 267